scripts/zones/Inner_Horutoto_Ruins/mobs/Goblin_Weaver.lua

```diff
@@ -2,10 +2,32 @@
 -- Area: Inner Horutoto Ruins
 --  Mob: Goblin Weaver
 -----------------------------------
+local ID = zones[xi.zone.INNER_HORUTOTO_RUINS]
+-----------------------------------
 local entity = {}
+local nmPopChance = 50
 
 entity.onMobDeath = function(mob, player, optParams)
     xi.regime.checkRegime(player, mob, 648, 2, xi.regime.type.GROUNDS)
 end
 
+entity.onMobDespawn = function(mob)
+    local totd = VanadielTOTD()
+    if totd == xi.time.NIGHT or totd == xi.time.MIDNIGHT then
+        if math.random(1, 100) <= nmPopChance then
+            local mobId = mob:getID()
+            local nmId = ID.mob.MAGICKED_BONES_PH_DAGGER[mobId]
+            if nmId then
+                local skele = GetMobByID(nmId)
+                local respawnTime = mob:getRespawnTime()
+                DisallowRespawn(mobId, true)
+                mob:timer(respawnTime, function()
+                    skele:setModelId(1096) --Dagger Skeleton Model ID
```

Comment:
I believe you could change the name of the two magicked bones in the mob_spawn_points table, and set the modelId onMobInitialize

Reply:
Potentially, I think the spawn point table is the only one where they have been distinct, otherwise both of them have a shared db entry in other tables so was trying to avoid doing the model changes via the db entries in particular

Reply:
yep, the first column is the lua file that gets read:

![image](https://github.com/LandSandBoat/server/assets/131182600/e9cd40a1-4cd8-414a-9127-dfe4a67d0bfc)

Scratch renaming it though just put an pair of if checks in the `onMobInitialize` function imo.

Reply:
Ahhh so that is how it determines which file to tie it to. I was asking about that in Discord for like 2 days >:( But yeah that could definitely be done there then for sure. I wasn't 100% sure if OnInitialize happened each time or exactly how it functioned

Reply:
How this gets tackled depends on the answers to some question first, because we want to do what retail does. And we normally would NOT got messing with the mobs model in script. I would straight reject a pr that did that without first proving a need to.

DOES retail let BOTH of these bones pop as either model, (as in, can you get 2 of them up at the same time both wielding same weapon) or does it just do one of each and you have to hope it rolls the dagger model one to get it? I suspect the latter. The answer to this will form how I advise you and what further questions I ask.

Reply:
I am not sure exactly how retail does it since I can't sub at the moment, the information that is based on is from the jp wiki (https://wiki.ffo.jp/html/7918.html https://wiki.ffo.jp/html/24263.html these pages specifically), where it stated (like I said the browser is kind of rough at translating it well so grain of salt) that each skeleton had a specific model based on which of the 2 goblins it popped from. In case your translation is different/better, here is exactly what it says for mine: "More specifically,[ Goblin Thug ](https://wiki.ffo.jp/html/10186.html)will be drawn as a one-handed club wielder, and Goblin Weaver a dagger will be drawn as wielding bone in the south hidden passageway entered from H-9"

As for  the drop, based on FFXIDB it only has a single entry for Magicked Bones for the zone, which has the dagger as part of the pool so I was operating under the assumption that they both could drop it. In general these 2 bones seem VERY unique among just about every other mob in the game for some reason.

Reply:
According to the sources the mob that spawns from the weaver has one model, and the thug has another model. If we don't set the model in the lua we just need 2 different mob_pools with everything except the model the same

Reply:
> I am not sure exactly how retail does it since I can't sub at the moment,

Not to be rude, but we're not guessing at things here or simply taking websites info at face value. I can possibly check for you myself over the weekend but if not there are folks in the captures discord who might be willing to spend some time on this. I'm a bit frazzled atm so I hope I am not coming off to much of a jerk. We need to know for sure when we implement or risk repeating the mistakes of previous iterations of the project. "I'll fix that later" and other jokes people told themselves.

> As for the drop, based on FFXIDB it only has a single entry for Magicked Bones for the zone, which has the dagger as part of the pool so I was operating under the assumption that they both could drop it. In general these 2 bones seem VERY unique among just about every other mob in the game for some reason.

That site will combine them for having the same name while being in the same zone. Their method of data capture can't tell the 2 apart.

Reply:
> According to the sources the mob that spawns from the weaver has one model, and the thug has another model. If we don't set the model in the lua we just need 2 different mob_pools with everything except the model the same

no you'd need 2 groups 2, in order to set 2 drop IDs. The dagger only drops from the one using the dagger model.

Reply:
Not coming off as a jerk, I get it. I am generally trying to use as many sources as I can to corroborate stuff the best I can currently.  I will have to join the captures discord it sounds like! If you end up finding the info as well and let me know, I can get it fixed pretty quick in terms of being retail accurate. Besides that though, it seems like mostly just reverting back to the original ph code with some extras to account for having to use spawnType 128

Reply:
FYI jp wiki indicates that both types of magicked bones models drop the dagger:
![image](https://github.com/LandSandBoat/server/assets/131182600/e73694d0-4eab-4734-93dc-eb5dcc704db5)

I think i've said all I can in this PR now, though. If we still require verification from retail it sounds like it shouldn't be that hard to get the mobs to pop, and hopefully not too hard to get the dagger to drop from the club-wielder.

Reply:
👍 I sometimes have a hard time gauging my own tone and also I _know_ I'm grumpy rn, so I was worried.

Spawn at night with PH being a mob that does NOT may add some extra trouble to this. Citipati's PH's are corses, who only pop at night. Multiple different values will just get treated as "scripted" and simply be overwritten (for a good migraine, take a look at what the functions toggle being allowed to spawn actually do to the spawnType field!) so combining other types with scripted doesn't work well. Past editors have made that mistake, as you found in Citipati.

I can possibly mod the NM function to handle night only as an optional thing. We probably don't want to check nighttime before rolling the chance to spawn because retail would check that at pop time - kill goblin just before night falls - as long as its "down" when night time hits, the spawn that occurs at night has the chance.

I've been considering adding a table param to it anyway and moving the force spawn option inside of that along with making the updatespawnpoint call optional.

Reply:
Yeah, the goblins being any time and the bones being night only has definitely been a source of annoyance for sure. I will see what I can manage with the current feedback though, and at least temporarily continuing with the assumption that they both can drop it until I can find a better source than the jp wiki stating as much. 

On that note though to accelerate testing, do you have any suggestions for advancing time that doesn't completely break things? Using !time both desyncs the local clock, seems to desync more over time, and makes VanadielTOTD return 0 until a natural TOTD change happens which was breaking my TOTD checks. Otherwise to test spawning I was just disabling the totd requirement to make sure other stuff behaved properly, then enabling it when time was getting close to try to emulate proper spawn conditions.

Reply:
> and at least temporarily continuing with the assumption that they both can drop it until I can find a better source than the jp wiki stating as much.

before SE added those lv80+ mobs I was the primary supplier of ceremonial daggers on my server. I know without any doubt that before they changed it, only the bones with rusty knife model coudl drop the dagger. The club model would not and the staff model would not (all 3 models existed in the zone in the past).

> On that note though to accelerate testing, do you have any suggestions for advancing time that doesn't completely break things? Using !time both desyncs the local clock, seems to desync more over time, and makes VanadielTOTD return 0 until a natural TOTD change happens which was breaking my TOTD checks.

Nope, its buggy while technically not being a bug, by its very nature. The game was never designed by SE to handle the time warpage so bad things happen unavoidably from us doing it. :( pity the in universe "time" wasn't instanced to each player/party so we coudl have timewarp spells/items, but that's a tall ask for an MMP especially back in the era FFXI was born in. Instancing wasn't even a thing yet! Some more modern games _literally every zone outside town is in an instance_ o.O